### PR TITLE
[feature](Nereids): infer not null from Agg Count() without groupby.

### PR DESCRIPTION
**Pull request by jackwener**
# Proposed changes

Issue Number: close #xxx

## Problem summary

infer not null from Agg Count()/Sum()..... without groupby.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.58 seconds
 stream load tsv:          431 seconds loaded 74807831229 Bytes, about 165 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          72 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230413032527_clickbench_pr_128322.html

**Comment by jackwener** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/InferAggNotNull.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.stream.Collectors;
+
+/**
+ * InferNotNull from Agg count(distinct);
+ */
+public class InferAggNotNull extends OneRewriteRuleFactory {
+    @Override
+    public Rule build() {
+        return logicalAggregate()
+                .when(agg -> agg.getAggregateFunctions().stream().allMatch(f -> f instanceof Count && f.isDistinct()))
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/InferAggNotNull.java:41]: why must distinct?

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall.

**Review by morrySnow — DISMISSED** [review]

**Comment by jackwener** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**